pkg/auth/audit/audit.go

```diff
@@ -94,12 +108,23 @@ func newAuditLog(writer *LogWriter, req *http.Request) (*auditLog, error) {
 	}
 
 	contentType := req.Header.Get("Content-Type")
-	if writer.Level >= levelRequest && bodyMethods[req.Method] && contentType == contentTypeJSON {
-		reqBody, err := readBodyWithoutLosingContent(req)
-		if err != nil {
-			return nil, err
+	loginReq := isLoginRequest(req.RequestURI)
+	if writer.Level >= levelRequest || loginReq {
+		if bodyMethods[req.Method] && contentType == contentTypeJSON {
+			reqBody, err := readBodyWithoutLosingContent(req)
+			if err != nil {
+				return nil, err
+			}
+			if loginReq {
+				loginName := getUserNameForBasicLogin(reqBody)
+				if loginName != "" {
+					auditLog.log.UserLoginName = loginName
+				}
+			}
+			if writer.Level >= levelRequest {
+				auditLog.reqBody = reqBody
+			}
 		}
```

Comment:
can we restructure this? e.g.
```go
	if exists, ok := bodyMethods[req.Method]; !exists || !ok {
		return auditLog, nil
	}
	if contentType != contentTypeJSON {
		return auditLog, nil
	}
	if writer.Level >= levelRequest {
                var err error
		auditLog.reqBody, err = readBodyWithoutLosingContent(req)
		if err != nil {
			return nil, err
		}
	}
	loginReq := isLoginRequest(req.RequestURI)
	if loginReq {
		auditLog.log.UserLoginName = getUserNameForBasicLogin(reqBody)
	}
	return auditLog, nil
```

Reply:
no,  since we need to readBodyWithoutLosingContent in both cases  when it is  levelRequest || loginReq, and we need to ensure the UserLoginName is set only for loginReq irrespective of the writer.Level